src/js/server/middleware.js

```diff
@@ -0,0 +1,92 @@
+// "Gatekeeper" middleware that rejects any incoming payloads that are not valid according to the schema provided in
+// `options.schemaContent`.  You can either pass the full content of the schema directly to `options.schemaContent` or
+// indicate the name of a schema file by setting `options.schemaFile` to the full path to your schema file.
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+require("./handler");
+
+fluid.defaults("gpii.schema.middleware.handler", {
+    gradeNames: ["gpii.express.handler"],
+    invokers: {
+        handleRequest: {
+            func: "{that}.sendResponse",
+            args: ["{that}.options.statusCode", "{that}.options.message"]
+        }
+    }
+});
+
+fluid.registerNamespace("gpii.schema.middleware");
+
+gpii.schema.middleware.getMiddleware = function (that) {
+    return that.rejectOrForward;
+};
+
+gpii.schema.middleware.rejectOrForward  = function (that, req, res, next) {
+    if (that.options.schemaContent) {
+        var results = that.validator.validate("schema", req.body);
+        if (results) {
+            // Instantiate a handler that will take care of the rest of the request.
+            that.events.onInvalidRequest.fire(req, res, 500, results);
+        }
+    }
+    else {
+        // We choose to fail if we can't validate the body.  That way anything downstream is guaranteed to
+        // only receive 2valid content.
```

Comment:
Yes, I was cleaning up lint errors, "F2" is the key for "next lint error", and I generally miss and hit "2" a dozen times in a cleanup session.